projects/packages/connection/src/class-manager.php

```diff
@@ -161,32 +160,27 @@ public static function configure() {
 	 * Sets up the XMLRPC request handlers.
 	 *
 	 * @since 1.25.0 Deprecate $is_active param.
+	 * @since $$next-version$$ Deprecate $request_params param.
 	 *
-	 * @param array                 $request_params incoming request parameters.
 	 * @param bool                  $has_connected_owner Whether the site has a connected owner.
 	 * @param bool                  $is_signed whether the signature check has been successful.
 	 * @param Jetpack_XMLRPC_Server $xmlrpc_server (optional) an instance of the server to use instead of instantiating a new one.
 	 */
 	public function setup_xmlrpc_handlers(
-		$request_params,
 		$has_connected_owner,
 		$is_signed,
 		Jetpack_XMLRPC_Server $xmlrpc_server = null
 	) {
 		add_filter( 'xmlrpc_blog_options', array( $this, 'xmlrpc_options' ), 1000, 2 );
 
-		if (
-			! isset( $request_params['for'] )
-			|| 'jetpack' !== $request_params['for']
-		) {
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended -- We are using the 'for' request param to early return unless it's 'jetpack'.
+		if ( ! isset( $_GET['for'] ) || 'jetpack' !== $_GET['for'] ) {
 			return false;
 		}
 
 		// Alternate XML-RPC, via ?for=jetpack&jetpack=comms.
-		if (
-			isset( $request_params['jetpack'] )
-			&& 'comms' === $request_params['jetpack']
-		) {
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended -- We are using the 'jetpack' request param to handle this request as an XML-RPC request if it's set to 'comms'.
```

Comment:
Theis explanation does not make it clear to me why it's safe.

The real question to answer is this: If some attacker tricked a site admin into following a link with `?for=jetpack&jetpack=comms` (and maybe other parameters), why will that not do anything bad?

p9dueE-4z8-p2 has more detail on that, with some common cases where it's ok and where it's not.

For example, if "This just determines whether to handle the request as an XML-RPC request. The actual XML-RPC endpoints do the appropriate nonce checking." is an accurate statement, that would be good IMO.

Reply:
Do you think it's more accurate to mention that the `alternate_xmlrpc` handler for the `template_redirect` action will [empty](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/connection/src/class-manager.php#L260) the `$_COOKIE` global anyways?

Reply:
Hmm, maybe. I see the call to `$this->require_jetpack_authentication()` slightly further down also does that.

Is there any possibility that something earlier in execution set `$current_user` or set any other authn- or authz-related stuff based on the cookies before we clear them?

Reply:
I added a combination of both your suggestion and mentioned the fact that we clear cookies later in method via `require_jetpack_authentication `.

> Is there any possibility that something earlier in execution set $current_user or set any other authn- or authz-related stuff based on the cookies before we clear them?

There shouldn't. This is executed very early on `plugins_loaded` via the [Config page](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/config/src/class-config.php#L75)